### PR TITLE
FPN

### PR DESCRIPTION
**Pull request by Dref360**
Feature Pyramid Network.

Unet with multi-scale prediction basically. CVPR 2017

### CODE REVIEW

File: deep-learning/_posts/2017-04-04-feature-pyramid-network.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+---
+
+The Feature Pyramid Network (FPN) looks a lot like the [U-net]({{ site.baseurl }}{% link deep-learning/_posts/2017-02-27-unet.md %}). The main difference is that there is multiple prediction layers: one for each upsampling layer.
+<div align="middle">
+  <img src="/deep-learning/images/fpn/architecture.png" width="400">
+</div>
+
+
+## FPN for region-proposal
+To achieve region-proposal, the authors add a 3x3 Conv layer followed by two 1x1 Conv for classification and regression on each upsampling layer. These additions are called **heads** and the weights are shared. For each head, you assign the sames anchors boxes, resized to match the head's shape. The ground truth labels are assigned to the anchor if it has >70% IoU.
```
Comment (lemairecarl) [inline deep-learning/_posts/2017-04-04-feature-pyramid-network.md]: > For each head, you assign the sames anchors boxes, resized to match the head's shape. The ground truth labels are assigned to the anchor if it has >70% IoU.

Could you clarify this part? Same anchor boxes as what? What does assigning a _ground truth_ label to an anchor box mean?

Also, "sames" should be "same".

File: deep-learning/_posts/2017-04-04-feature-pyramid-network.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+## FPN for object Detection
+Using Fast(er) R-CNN, they can use FPN as the region proposal part. The proposals are used in combination with RoiPooling and then they can do the same works as Fast(er) R-CNN.
```
Comment (lemairecarl) [inline deep-learning/_posts/2017-04-04-feature-pyramid-network.md]: same work

File: deep-learning/_posts/2017-04-04-feature-pyramid-network.md [hunk 3]
```diff
@@ -9,18 +9,19 @@ cite:
   title:   "Feature Pyramid Networks for Object Detection"
   venue:   "Computer Vision and Pattern Recognition, 2017. CVPR 2017"
 ---
-
+s
```
Comment (lemairecarl) [inline deep-learning/_posts/2017-04-04-feature-pyramid-network.md]: <kbd>CTRL</kbd>+<kbd>S</kbd>

File: deep-learning/_posts/2017-04-04-feature-pyramid-network.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+tags:   deep-learning CNN localization bounding-boxes
+author: Frédéric Branchaud-Charron
+pdf:   "https://arxiv.org/pdf/1612.03144.pdf"
+cite:
+  authors: "T. Lin, P. Dollár, R. Girshick, K. He, B. Hariharan, S. Belongie"
+  title:   "Feature Pyramid Networks for Object Detection"
+  venue:   "Computer Vision and Pattern Recognition, 2017. CVPR 2017"
+---
+
+The Feature Pyramid Network (FPN) looks a lot like the [U-net]({{ site.baseurl }}{% link deep-learning/_posts/2017-02-27-unet.md %}). The main difference is that there is multiple prediction layers: one for each upsampling layer.
```
Comment (jodoin) [inline deep-learning/_posts/2017-04-04-feature-pyramid-network.md]: Is the lateral connections also a big difference?

**Pull request merged**